docs/how_to/create_a_map.rst

```diff
@@ -0,0 +1,99 @@
+.. _how-to-create-a-map:
+
+How to create a sunpy Map
+=========================
+
+One of the primary goals of the Map interface is to make it as easy as possible to create a Map.
+As such, you can pass many different kinds of inputs to Map.
+These are enumerated below.
+
+File name
+---------
+
+This is the easiest and recommended way to create a Map. This can be either a string or a `pathlib.Path`.
+
+.. code-block:: python
+
+    >>> import sunpy.map
+    >>> import sunpy.data.sample
+    >>> my_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)  # doctest: +REMOTE_DATA
+    >>> my_map = sunpy.map.Map('file.fits')  # doctest: +SKIP
+    >>> import pathlib
+    >>> my_map = sunpy.map.Map(pathlib.Path('file.fits'))  # doctest: +SKIP
+    >>> sub_dir = pathlib.Path('local_dir/sub_dir')
+    >>> mymap = sunpy.map.Map(sub_dir / 'another_file.fits')   # doctest: +SKIP
```

Comment:
```suggestion
    >>> import pathlib
    >>> import sunpy.map
    >>> import sunpy.data.sample

    >>> my_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)  # doctest: +REMOTE_DATA
    >>> my_map = sunpy.map.Map('file.fits')  # doctest: +SKIP
    >>> my_map = sunpy.map.Map(pathlib.Path('file.fits'))  # doctest: +SKIP
    >>> sub_dir = pathlib.Path('local_dir/sub_dir')
    >>> mymap = sunpy.map.Map(sub_dir / 'another_file.fits')   # doctest: +SKIP
```
I feel like a line break is ok?

Reply:
I'm not a fan of imposing this within the how-to or in sample code in general.

Reply:
I agree placing the import statements above the code is preferable. I don't like the spacing requirements that we typically impose in the rest of the codebase.